WireAPI/Tests/WireAPITests/Helpers/RequestSnapshotHelper.swift

```diff
@@ -52,6 +52,23 @@ struct RequestSnapshotHelper {
         }
     }
 
+    func verifyRequestForAllAPIVersions(
+        when block: (any UsersAPI) async throws -> Void,
+        file: StaticString = #file,
+        function: String = #function,
+        line: UInt = #line
+    ) async throws {
+        for apiVersion in APIVersion.allCases {
+            try await verifyRequest(
+                apiVersion: apiVersion,
+                when: block,
+                file: file,
+                function: function,
+                line: line
+            )
+        }
+    }
```

Comment:
It would be nice to try make these verify methods generic, so we avoid the duplication each time we need to test a different api.

Reply:
What if we pass a build function inside the init of `SnapshotHelper`? We could then create the helper once in the test setup and not need to clutter the assertion methods.

Reply:
Good idea!

Reply:
I've been looking into that clunky generic too 😆 What do you think about this?
```swift
struct RequestSnapshotHelper2<Builder: APIBuilder> {

    var makeBuilder: (any HTTPClient) -> Builder

    // ...other implementation...

    @MainActor
    func verifyRequest(
        apiVersion: APIVersion,
        onPerform block: (Builder.API) async throws -> Void,
        file: StaticString = #file,
        function: String = #function,
        line: UInt = #line
    ) async throws {
        let httpClient = HTTPClientMock()
        let apiBuilder = makeBuilder(httpClient)
        let api = apiBuilder.makeAPI(for: apiVersion)
        try? await block(api)
        let request = try XCTUnwrap(httpClient.receivedRequest, "no request was generated")

        let errorMessage = verifySnapshot(
            of: request,
            as: .dump,
            named: "v\(apiVersion.rawValue)",
            file: file,
            testName: function,
            line: line
        )

        if let errorMessage {
            XCTFail(errorMessage, file: file, line: line)
        }
    }
}
```

And usage:
```swift
    var snapshotHelper: RequestSnapshotHelper2<UsersAPIBuilder>!

    override func setUp() {
        super.setUp()
        snapshotHelper = .init() { UsersAPIBuilder(httpClient: $0) }
    }

    func testGetUserRequest() async throws {
        try await snapshotHelper.verifyRequestForAllAPIVersions { api in
            _ = try await api.getUser(for: .mockID1)
        }
    }
```

Reply:
Much better!

Reply:
Integrated this